Fix problem with NaN values being generated during absorption correction in PEARL scripts

DannyHindson opened the pull request:
**Description of work.**

Fix problem with NaN values being generated during absorption correction when absorption workspace is different size to the Vanadium workspace. This was causing a crash later on in the reduction process in a call to FindPeaks

Occurs on PEARL runs from 19_1 and 19_2 cycle where HRPD detector cards used

The fix is to remove spectra from the absorption correction and V workspaces so they contain a common set of spectra

**To test:**

Unzip the file \\olympic\Babylon5\Public\DannyHindson\PR32759_files.zip
Update the script below to replace <folder> with the path to the folder where you unzipped the file
Run this script and it should complete without an error. You'll need access to the ISIS data archive to run this:

```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from isis_powder.pearl import Pearl

pearl_focus = Pearl(user_name="Calib",
                    output_directory=r"<folder>\output",   #directory above cycle number 
                    config_file=r"<folder>\Calibration.yaml",
                    calibration_directory=r"<folder>\calib_critical_files",
                    calibration_mapping_file=r"<folder>\pearl.yaml")

pearl_focus.create_vanadium(run_in_cycle=110388,tt_mode="tt35",do_absorb_corrections=True,long_mode=True)
pearl_focus.create_vanadium(run_in_cycle=110388,tt_mode="tt70",do_absorb_corrections=True,long_mode=True)
pearl_focus.create_vanadium(run_in_cycle=110388,tt_mode="tt88",do_absorb_corrections=True,long_mode=True)
```

Fixes #32568.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DannyHindson committed:
Fix problem with NaN values being generated during abs corr

...when absorption workspace is different size to the Vanadium workspace
Occurs on PEARL runs from 19_1 and 19_2 cycle where HRPD detector cards
used

RichardWaiteSTFC reviewed (changes requested):
Nice fix, will approve when set intersection added as discussed offline

DannyHindson committed:
Couple of improvements

Add extra check if overlap between workspaces is zero
When checking if overlap detector list is same as the sample list
temporarily sort the lists first (ditto for similar check on V det list)

DannyHindson committed:
Revert change to sort det id lists

ExtractSpectra resorts a workspace if the detector id list contains the
same detector ids as are present in the input workspace but they're sorted
differently. This is a useful feature if the V and sample workspace happen
to be sorted differently so it makes sense to allow this to happen.
Comparing the common_det_ids to the V and sample detector id lists without
sorting will pick up any sort order differences

RichardWaiteSTFC reviewed (approved):
Nice fix, script executes without error and I can see the message in the log.
